### DISCUSSION TITLE
pass `params` and `searchParams` into template files

### DISCUSSION THREAD

**Discussion by PHILLIPS71** [vercel/next.js#70952] [Ideas]
### Goals

1. adds `params` and `searchParams` to the template.js file's props 

### Background

It's understandable that the layout file doesn't receive route parameters since layouts aren't re-rendered during navigation. However, from what I understand, templates are re-rendered between routes, so it would make sense for route parameters to be passed to the template.

It simplifies situations where you need to render content based on route parameters, and handling it on the server offers a much better user experience. It's also far easier, as it eliminates concerns about managing loading states or content skeletons.

### Proposal

It should work like the page file, where both `params` and `searchParams` are passed in.

**Comment by renekahr** [comment]
I don't see any reason why template wouldn't receive the searchParams prop. I thought this is the whole point compared to a layout.
